internal/infrastructure/router/public_test.go

```diff
@@ -437,3 +387,26 @@ func TestSkipperMixedPredicate(t *testing.T) {
 		assert.Equal(t, testCase.Expected, result)
 	}
 }
+
+func TestGuardNewGroupPublic(t *testing.T) {
```

Comment:
usually guard tests use `assert.PanicsWithValue` but here is only 4 times:   `assert.Panics` so it is not clear if the right/expected panic is caught.